lib/runtime/wasmer/exports.go

```diff
@@ -8,26 +8,25 @@ import (
 
 	"github.com/ChainSafe/gossamer/dot/types"
 	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/runtime/errors"
 	"github.com/ChainSafe/gossamer/lib/transaction"
 	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
 // ValidateTransaction runs the extrinsic through the runtime function
-// TaggedTransactionQueue_validate_transaction and returns *Validity
-func (in *Instance) ValidateTransaction(e types.Extrinsic) (*transaction.Validity, error) {
+// TaggedTransactionQueue_validate_transaction and returns *Validity. Two types of errors
+// are returned here: 1) *txnvalidity.TransactionValidityError is a VDT containing
+// VDTs of the types of transaction validity errors. The whole VDT is returned so
+// the caller can handle as it seems fit, as this will vary per use case. 2) normal error
+// returned if something fails in the process i.e. unmarshalling error
+func (in *Instance) ValidateTransaction(e types.Extrinsic) (
+	*transaction.Validity, error) {
 	ret, err := in.Exec(runtime.TaggedTransactionQueueValidateTransaction, e)
 	if err != nil {
 		return nil, err
 	}
 
-	if ret[0] != 0 {
-		return nil, runtime.NewValidateTransactionError(ret)
-	}
-
-	v := transaction.NewValidity(0, [][]byte{{}}, [][]byte{{}}, 0, false)
-	err = scale.Unmarshal(ret[1:], v)
-
-	return v, err
+	return errors.UnmarshalTransactionValidity(ret)
```

Comment:
Can you move this new `errors` package to `internal/runtime/errors` please? Trying to add newer packages in there 😉 Especially since it's only used in internal runtime code.

Reply:
Resolving this for now, we'll get back to where to place this code once other conversations are resolved.